src/lib.rs

```diff
@@ -203,7 +209,7 @@ impl ChallengeResponse {
         let mut q_params = String::new();
 
         match nonce {
-            Nonce::Value(val) if val.len() > 0 => {
+            Nonce::Value(val) if val.is_empty() => {
```

Comment:
I don't think this is what you want

Reply:
Unfortunately `clippy` keeps suggesting this, I find `val.len() > 0` more intuitive than `!val.is_empty()`, but ohwell...